### ISSUE TITLE
trying to import ~300MB sql file

### ISSUE THREAD

**Issue by Gavrisimo** [TablePlus/DBngin#14]
1. Driver (Ex: PostgreSQL 10.0):

MySQL 5.7.23

2. DBngin build number:

1.0 (14)

3. macOS version:

10.14.1 (18B75)

4. The steps to reproduce this issue:

- start mysql service
- connect to it through tableplus
- open database
- file -> import -> from sql dump
- select ~300MB sql file
- start import
- tableplus crashes

I am submitting this here because when I tried to run the "restore backup" or just simply copy/past entire content of the file to the sql editor and running it like that, in both those cases I see message like "MySQL has gone away" or something like that.

This makes me think that the MySQL server dies during the import which makes the tableplus crash.

[tableplus-crash-report.txt](https://github.com/TablePlus/DBngin/files/2576366/tableplus-crash-report.txt)

**Comment by huyphams** [comment]
@Gavrisimo which TablePlus build are you using? 300 MB shouldn’t be a problem, I have tested with a GB file dump.

**Comment by huyphams** [comment]
Yup, I’m using DBngin and TablePlus, looks like it was an other issue. Lets me see the crash log.

**Comment by Gavrisimo** [comment]
If you want I can email you link to download database?

**Comment by huyphams** [comment]
Sure, that’s awesome, thank you so much.
My email is huy@tableplusapp.com 👍

**Comment by huyphams** [comment]
Update: I got the bug:
```
(2018-11-13 20:42:22.902330+0700 TablePlus[855:20195] [General] An uncaught exception was raised
4910, 2233, 2018-11-13 20:42:22.902375+0700 TablePlus[855:20195] [General] NSWindow drag regions should only be invalidated on the Main Thread!
```

Looks like it was a UI issue when TablePlus tries to show a popup (probably an error popup). I'm fixing it now. it's so easy to reproduce with your dump 👍

**Comment by Gavrisimo** [comment]
Yeah, my dumps are the best... ¯\_(ツ)_/¯

**Comment by Gavrisimo** [comment]
Just FYI when I can import that DB dump without any issues when importing to MySQL running on my virtual machine.

**Comment by huyphams** [comment]
Yeah maybe it has bigger @max_allowed_packet, I'm reading the MySQL document about this threshold and see if we can change it safely.

**Comment by huyphams** [comment]
It's fixed https://www.dropbox.com/s/0i0ofnhi88vlxz8/TablePlus.dmg?dl=0
In the meantime, you can always use backup/restore function to perform restore a dump file in MySQL.

**Comment by huyphams** [comment]
I released the build 154 which includes this fixes.

**Comment by Gavrisimo** [comment]
> It's fixed https://www.dropbox.com/s/0i0ofnhi88vlxz8/TablePlus.dmg?dl=0
> In the meantime, you can always use backup/restore function to perform restore a dump file in MySQL.

Yeah, I know... The thing is I need to export the database from WordPress admin panel using migrate db pro plugin, so I have a functional DB in my local environment... :-/

**Comment by huyphams** [comment]
I mean, the backup/restore function is supported by TablePlus
<img width="249" alt="screen shot 2018-11-14 at 4 00 15 pm" src="https://user-images.githubusercontent.com/3654622/48471388-6377d880-e826-11e8-89d3-50502cf5c508.png">

**Comment by Gavrisimo** [comment]
I know! :) The thing with these WordPress DBs is that it's the worst...

You can't just make a backup or SQL dump from production DB and restore/import it to local environment using a tool like tableplus or terminal or whatever...

The issue is that they've got the URL of the production website inside the database *all over* the place. In bunch of places the data is saved as serialized data using php's `serialize()` function and updating serialized data has its set of issues and challenges...

That's why plugins like `WP Migrate DB Pro` help a lot, where you just enter production URL and local dev URL and it exports the database all ready to go, you just need to import it...

Basically for all other projects I use backup/restore through tableplus, but the WordPress ones are weird... :-/

**Comment by huyphams** [comment]
I got it, it I have tried to restore the file you sent me via restore, it works. So you don't need to use the backup but can use the restore function.

**Comment by Gavrisimo** [comment]
If I try to use "restore" with this dump I have sent you I get this:

![image](https://user-images.githubusercontent.com/203706/48472015-90f85e00-e7f5-11e8-9e03-92350d345422.png)

And if I try to use "import" function I get this:

![image](https://user-images.githubusercontent.com/203706/48472063-b38a7700-e7f5-11e8-919b-d6a4511e6493.png)

I tried running that SQL and importing it again, but it failed with the same error.

I then reconnected to the MySQL instance started importing it again and it seems like it is working!!

So I stopped it just to test restarting the MySQL instance and trying the import again, but that just brought me back to begging with that `max_allowed_packet` error.

It seems that running `SET GLOBAL max_allowed_packet=1073741824;` is not being preserved after MySQL restart.

**Comment by huyphams** [comment]
Yeah, I think it has the same error, but the backup/restore was implemented by Oracle (the maintainer of MySQL), they didn't add the structure warning about the max allowed package.

The `restore from dump` was implemented by me and I have added the instructions to tell the user what they can do to resolve the issue.

After increasing the max allowed package, I believe that both the `backup/restore` and `restore from dump` functions will work fine.
Thanks for testing 👍
